cea/utilities/epwreader.py

```diff
@@ -37,29 +38,28 @@ def epw_reader(weather_path):
 
     year = epw_data["year"][0]
     # Create date range from epw data
-    date_range = pd.DatetimeIndex(pd.to_datetime(dict(year=epw_data.year, month=epw_data.month, day=epw_data.day, hour=epw_data.hour-1))
+    date_range = pd.DatetimeIndex(
+        pd.to_datetime(dict(year=epw_data.year, month=epw_data.month, day=epw_data.day, hour=epw_data.hour - 1))
     )
+    epw_data['date'] = date_range
+    epw_data['dayofyear'] = date_range.dayofyear
     if isleap(year):
-        if epw_data.shape[0] == (HOURS_IN_YEAR+24):
-            epw_data['date'] = date_range
-            epw_data['dayofyear'] = date_range.dayofyear
-            epw_data = result[~((date_range.month == 2) & (date_range.day == 29))].reset_index()
-        elif epw_data.shape[0] == HOURS_IN_YEAR:
-            print("you have a leap year, but the weather file was modified already, we account for this "
-                  "no need of further action")
-            date_range = date_range[~((date_range.month == 2) & (date_range.day == 29))]
-            epw_data['date'] = date_range
-            epw_data['dayofyear'] = date_range.dayofyear
-    else:
-        if len(epw_data) != HOURS_IN_YEAR:
-            raise Exception('Incorrect number of rows. Expected {}, got {}'.format(HOURS_IN_YEAR, len(epw_data)))
-        epw_data['date'] = date_range
-        epw_data['dayofyear'] = date_range.dayofyear
+        epw_data = epw_data[~((date_range.month == 2) & (date_range.day == 29))].reset_index()
```

Comment:
This line fails, since for a file with a leap year but without that day and month, it will fail. I sent you privately a file with that characteristics to test

Reply:
It should work already since I generate the datetime index based on the data from the epw and not using intervals previously.